### PR TITLE
Route for getting an array of colour names for a product

### PR DESCRIPTION
**Pull request by PhilipEnchin**

### CODE REVIEW

**Comment by JohnReiland** [comment]
I'm trying doing a review by the books (Learn):

**Phase 1:**

- Typos: I could object to the spelling of colour, but in this group I'd be outvoted. Other than that, no errors.
- Poor names for variables, functions, classes, etc: As an outsider, 'colours.colourName colours.colour' is a bit unclear. Could be a more explicit, but then again, that road can lead to obnoxiously long variable names, so take it just as a suggestion.
- Overly-large functions or classes: Don't believe so.
- At least one test: I only just started doing tests, so I don't think I can be one to talk.
- Cruft: None.
- Violations of HR's styleguide: None.

**Phase 2:** 

- If after taking the time to read whatever the PR links to, the reviewer can't actually understand what the code is doing, send it back for clarification: No trouble here.
- Feature spec omissions. None that occur to me at this time.
- Adequate test coverage for changed lines or critical code paths: N/A yet, I think.
- Tests assert valuable behaviors: N/A yet, I think.
- There are negative tests, not just happy-path tests: N/A yet, I think.
- Tests assert the interfaces of the components involved (their API's) -- NOT reaching into the internals to test implementation details: N/A yet, I think.
- Propose alternative implementations or refactoring that will increase readability, maintainability, extensibility: Maybe a few small comments describing in plain English what a block of code does, to make it easier for the person who didn't write it to see what it's doing? I don't know; I think I might need more clarification than most. Me not immediately seeing what code does shouldn't alarm anyone.
- Proposals for making functions do only one thing: Yours already do this.
- Proposals for removing side effects from functions: Don't believe yours does this.
- Proposals for factoring out helper functions or otherwise modularizing to make crisper "Lego blocks" to build the solution with: I don't believe this needs further deconstructing.
- Unhandled edge-cases based on assumptions that a component has about its data: How would this handle a request for a non-existent productId?
- Ways to reduce the number of things that a single component, especially a function, is doing: Again, not needed.

**Comment by PhilipEnchin** [comment]
Thanks folks. I'm going to handle non-existent products later. Right at this moment, I have only one of three routes on my master branch. Once they're all merged, I'll get on it.

**Comment by JohnReiland** [comment]
> Everything looks good, just 1 small typo: "colour"

Wait, do Englishmen not spell it colour??

**Comment by PhilipEnchin** [comment]
image
